### PR TITLE
Integrate BDP UI Library Footer into BDP Website

### PR DESCRIPTION
**Pull request by jrakibi**
This PR adds the footer component from the BDP UI library to the BDP website

https://www.npmjs.com/package/@bitcoin-dev-project/bdp-ui

### CODE REVIEW

File: components/footer-component.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+"use client"
+
+import React from "react"
+import { Footer } from "@bitcoin-dev-project/bdp-ui"
+import Link from "next/link"
+// import { menuApps } from "@/utils/data";
+import Image from "next/image"
+
+function Wrapper({
```
[the remark this answers is not in the document]
Comment (Emmanuel-Develops) [reply components/footer-component.tsx:9]: Curious did you use `max-md` due to conflicting styles?

**Comment by aassoiants** [comment]
Also to include: https://www.linkedin.com/company/bitcoin-dev-project/

File: components/footer-component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jrakibi) [reply components/footer-component.tsx:6]: - 'Good First Issue' and 'Decoding Bitcoin' use full-width layouts, which is why I've used the `Wrapper` component, similar to what we have for the transcript registry. I think using a full-width footer on the homepage looks fine, unless we decide to make it dynamic (using max-w-7xl on the homepage and full width for Decoding and Good First Issue
- I've  removed `max-md`

**Comment by jrakibi** [comment]
> Also to include: https://www.linkedin.com/company/bitcoin-dev-project/

@Emmanuel-Develops I think this should be added on the library level, right ? 
we don't have linkedin on the list

**Comment by Emmanuel-Develops** [comment]
> > Also to include: https://www.linkedin.com/company/bitcoin-dev-project/
> 
> @Emmanuel-Develops I think this should be added on the library level, right ? we don't have linkedin on the list

not necessarily, I made it very customizable so you can extend with custom SM e.g linkedIn doing something like this:
```
{
              entity: "linkedIn",
              entityLink: "https://www.linkedin.com/company/bitcoin-devs/",
              iconProps: {
                className: "hover:text-orange-400",
              },
              icon: <LinkedInIcon className="w-full h-full" />,
}
```
tldr: if you provide your own icon you can support any arbitrary SM entity.

I need to invest in documentation 😅

**Comment by jrakibi** [comment]
> > > Also to include: https://www.linkedin.com/company/bitcoin-dev-project/
> > 
> > 
> > @Emmanuel-Develops I think this should be added on the library level, right ? we don't have linkedin on the list
> 
> not necessarily, I made it very customizable so you can extend with custom SM e.g linkedIn doing something like this:
> 
> ```
> {
>               entity: "linkedIn",
>               entityLink: "https://www.linkedin.com/company/bitcoin-devs/",
>               iconProps: {
>                 className: "hover:text-orange-400",
>               },
>               icon: <LinkedInIcon className="w-full h-full" />,
> }
> ```
> 
> tldr: if you provide your own icon you can support any arbitrary SM entity.
> 
> I need to invest in documentation 😅

Thanks you, done

File: components/footer-component.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                                }
+                            },
+                            {
+                                entity: "linkedIn",
+                                entityLink:
+                                    "https://www.linkedin.com/company/bitcoin-devs/",
+                                iconProps: {
+                                    className: "hover:text-orange-400"
+                                },
+                                icon: <LinkedinIcon className="w-full h-full" />
```
Comment (Emmanuel-Develops) [inline components/footer-component.tsx:64]:
```suggestion
                                icon: <LinkedinIcon className="w-full h-full" />
                                icon: <LinkedinIcon width="100%" height="100%" />
```

**Review by Emmanuel-Develops — CHANGES_REQUESTED** [review]
The icon doesn't take the full width of the container
Screenshot 2024-11-15 at 4 03 17 PM

**Comment by Emmanuel-Develops** [comment]
also ideally you'd want to import the library CSS globally, so that custom classes and colors apply
```tsx
// app/layout.tsx
import { Barlow } from "next/font/google"
import type { Metadata } from "next"
import "@bitcoin-dev-project/bdp-ui/styles.css" // add the library styles here before global css
import "./globals.css"
...
```

**Review by Emmanuel-Develops — APPROVED** [review]
